### ISSUE TITLE
Rename repository to `interop-tests`

### ISSUE THREAD

**Issue by thomaseizinger** [libp2p/test-plans#109]
Suggestion to change `test-plans` to `interop-tests` everywhere. I think it is more expressive on what these tests are testing.

**Comment by MarcoPolo** [comment]
Agreed. Lower priority for me, but I like it.

To list what else would have to change:
- [ ] go-libp2p test-plan test (and rename to interop-tests as well)
- [ ] rust-libp2p test-plan test (and rename to interop-tests as well)

**Comment by p-shahi** [comment]
Does `interop-tests` cover the scope of planned future tests too? And if we're already testing stuff like expected RTT, maybe it should be `e2e-tests`

**Comment by thomaseizinger** [comment]
> Does `interop-tests` cover the scope of planned future tests too? And if we're already testing stuff like expected RTT, maybe it should be `e2e-tests`

Good point, I am okay with `e2e-tests` too.

**Comment by p-shahi** [comment]
Is doing this still worthwhile? I think it's a nice distinction to make and allows us to leave test-plans in the past but it also might be harder to pull off after in-repo testing

**Comment by marten-seemann** [comment]
Yes, GitHub creates a redirect, I recently went through this process when I migrated quic-go to its own org. It still makes sense to update implementations to use the correct path.

That said, are we getting a lot out of a renaming?

**Comment by thomaseizinger** [comment]
With the redirect, the good news is that we don't have to do anything in lockstep, thus my point would be that a more expressive name is worth it.

Let's set a deadline for end of June.

Regarding the actual name, what I most commonly hear from people in conversations is `interop-tests` and not `e2e-tests`. `interop-tests` for me suggests that we are testing interoperability whereas `e2e-tests` has more a focus on testing functionality.

I'd consider hole-punching tests more on the functional side rather than on the interop side.

Plus, with the addition of the performance benchmarks, one could even debate whether "tests" is the right category for all of this. Just `interop` perhaps? Even with the addition of hole-punching tests, I find `interop` to be a much more intriguing name compared to `e2e`.

**Comment by p-shahi** [comment]
`interop` sounds good to me. Shall we go ahead with the change.

> Let's set a deadline for end of June.

That sounds good.

> That said, are we getting a lot out of a renaming?

It is cosmetic but why keep referring to test-plans when Testground is no longer used. It also is more evident to newcomers (new implementation adding their interop tests).

**Comment by thomaseizinger** [comment]
`interop` seems to be already taken by the JS specific interop runner? :/

https://github.com/libp2p/interop
